corehq/apps/reports/static/reports/js/bootstrap5/case_details.js

```diff
@@ -72,20 +72,13 @@ hqDefine("reports/js/bootstrap5/case_details", [
 
         self.data_loading = ko.observable(false);
 
-        self.getParameterByName = function (name, url) {
-            if (!url) {
-                url = window.location.href;
+        self.getFormIdFromUrl = function () {
+            var regex = new RegExp("[?&]form_id(=([^&#]*)|&|#|$)"),
+                results = regex.exec(window.location.hash);
```

Comment:
They look like this:
`/a/domain/reports/case_data/928841d0-9376-45cc-bd22-7c6cd2069a27/#history?form_id=8daba6c041914e288c87aa35cfa5c7fe`
Honestly this regex looks a little cryptic to me, not entirely sure how it works, but since it does, I figured I wouldn't fiddle with it much (beyond in-lining `form_id`).
Looking at it a little closer, I believe it functions by `[^&#]*` matching non-delimiter characters.

Reply:
I had the same questions, actually - `url` was always provided, and it was set to `window.location.hash`, so the `href` business never came into play.  But yes, `hash` does seem like an odd choice - seems like `window.location.search` would be better :shrug: